hdInsight-apache-spark-1-4-1/azuredeploy.json

```diff
@@ -0,0 +1,226 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "allowedValues": [
+                "Central US",
+                "East Asia",
+                "East US",
+                "Japan East",
+                "Japan West",
+                "North Europe",
+                "South Central US",
+                "Southeast Asia",
+                "West Europe",
+                "West US",
+                "East US 2"
+            ],
+            "metadata": {
+                "description": "The location where all azure resources will be deployed."
+            }
+        },
+        "clusterName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the HDInsight cluster to create."
+            }
+        },
+        "LoginUsername": {
+            "type": "string",
+            "defaultValue": "hdiuser",
+            "metadata": {
+                "description": "These credentials can be used to submit jobs to the cluster, log into cluster dashboards, log into Ambari, and SSH into the cluster."
+            }
+        },
+        "LoginPassword": {
+            "type": "securestring",
+            "metadata": { 
+                "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
+            }
+        },
+        "clusterNodeSize": {
+            "type": "string",
+            "defaultValue": "Standard_D12",
+            "allowedValues": [
+                "Standard_D12",
+                "Standard_D13",
+                "Standard_D14"
+            ],
+            "metadata": {
+                "description": "All nodes will be deployed using the specified hardware profile: D12(4 CPU Cores, 28GB of RAM), D13(8 CPU Cores, 56GB of RAM), D14(16 CPU Cores, 112 GB of RAM)."
+            }
+        },
+        "clusterWorkerNodeCount": {
+            "type": "int",
+            "defaultValue": 2,
+            "metadata": {
+                "description": "The number of worker nodes in the HDInsight cluster."
+            }
+        }
+     },
+    "variables": {
+        "defaultApiVersion": "2015-05-01-preview",
+        "clusterApiVersion": "2015-03-01-preview",
+        "clusterType": "hadoop",
+        "clusterVnetSubnetName": "[concat(parameters('clusterName'),'-subnet')]",
+        "clusterVnetAddressSpace": "172.16.228.0/23",
+        "clusterVNetSubnetAddressRange": "172.16.229.0/24",
+        "clusterStorageAccountName": "[concat(parameters('clusterName'),'storeacc')]",
```

Comment:
There's a high possibility of template deployment failing due to globally unique storage account naming requirements not being met due to this concatenation. For e.g. two users deploying this template for the first time choose "test" as the name of their cluster, only one of them will succeed. 

It's better to get storage account and dns names as user entered parameters to satisfy the uniqueness requirements